robottelo/api/utils.py

```diff
@@ -293,3 +302,242 @@ def create_role_permissions(role, permissions_types_names):  # pragma: no cover
+            mask=settings.vlan_networking.netmask,
+            domain=[domain],
+            location=[loc],
+            organization=[org],
+            dns=proxy,
+            dhcp=proxy,
+            tftp=proxy,
+            discovery=proxy
+        ).create()
+
+    # Search if Libvirt compute-resource already exists
+    # If so, just update its relevant fields otherwise,
+    # Create new compute-resource with 'libvirt' provider.
+    resource_url = u'qemu+ssh://root@{0}/system'.format(
+        settings.compute_resources.libvirt_hostname
+    )
+    comp_res = [
+        res for res in entities.LibvirtComputeResource().search()
+        if res.provider == 'Libvirt' and res.url == resource_url
+    ]
+    if len(comp_res) >= 1:
+        computeresource = entities.LibvirtComputeResource(
+            id=comp_res[0].id).read()
+        computeresource.location.append(loc)
+        computeresource.organization.append(org)
+        computeresource = computeresource.update([
+            'location', 'organization'])
+    else:
+        # Create Libvirt compute-resource
+        computeresource = entities.LibvirtComputeResource(
+            provider=u'libvirt',
+            url=resource_url,
+            set_console_password=False,
+            display_type=u'VNC',
+            location=[loc.id],
+            organization=[org.id],
+        ).create()
+
+    # Get the Partition table ID
+    ptable = entities.PartitionTable().search(
+        query={
+            u'search': u'name="{0}"'.format(DEFAULT_PTABLE)
+        }
+    )[0].read()
+
+    # Get the OS ID
+    os = entities.OperatingSystem().search(query={
+        u'search': u'name="RedHat" AND (major="{0}" OR major="{1}")'
+        .format(RHEL_6_MAJOR_VERSION, RHEL_7_MAJOR_VERSION)
+        })[0].read()
+
+    # Get the Provisioning template_ID and update with OS, Org, Location
+    provisioning_template = entities.ConfigTemplate().search(
+        query={
+            u'search': u'name="{0}"'.format(DEFAULT_TEMPLATE)
+        }
+    )
+    provisioning_template = provisioning_template[0].read()
+    provisioning_template.operatingsystem.append(os)
+    provisioning_template.organization.append(org)
+    provisioning_template.location.append(loc)
+    provisioning_template = provisioning_template.update([
+        'location',
+        'operatingsystem',
+        'organization'
+    ])
+
+    # Get the PXE template ID and update with OS, Org, location
+    pxe_template = entities.ConfigTemplate().search(
+        query={
+            u'search': u'name="{0}"'.format(DEFAULT_PXE_TEMPLATE)
+        }
+    )
+    pxe_template = pxe_template[0].read()
+    pxe_template.operatingsystem.append(os)
+    pxe_template.organization.append(org)
+    pxe_template.location.append(loc)
+    pxe_template = pxe_template.update(
+        ['location', 'operatingsystem', 'organization']
+    )
+
+    # Get the arch ID
+    arch = entities.Architecture().search(
+        query={u'search': u'name="x86_64"'}
```

Comment:
we have `DEFAULT_ARCHITECTURE = 'x86_64'` in master. Maybe, you like to move that change to 6.2.z